Improve ReadMes and add generate scenario script (#3)
inductiva/sph-demos: SPH_SIMULATIONS.md

## Running an SPH Simulation

> For **power users** who know how to configure a specific simulator and prepare the respective simulation files, **Inductiva API** provides a simple interface to simulate on a large scale. Here, we demonstrate how to use two SPH simulators in **Inductiva API**.

Reviewer: ```suggestion
This section assumes users know how to configure a SplishSplash and/or DualPhysics simulation,
in which case our **Inductiva API client** provides a simple interface to run those simulations at
scale. Users are, otherwise, invited to consult the relevant documentation for each of those
simulation engines (LINKS HERE and HERE, respectively).
```

Author: While I understand your point of setting  **Inductiva API client**, I think keeping it simple as it was is more clean and still includes everything.

Further, in the new line 5, `otherwise` seems contradictory. I think "Users are, in any case, invited..."

## Running an SPH Simulation

This section assumes users know how to configure a SplishSplash and/or DualSPHysics
simulation, in which case our **Inductiva API client** provides a simple interface to run
those simulations at scale. Users are, in any case, invited to consult the relevant documentation
for each of those simulation engines (links
[here](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH) and
[here](https://github.com/DualSPHysics/DualSPHysics), respectively). 

Let's demonstrate how to use two SPH simulators with **Inductiva API**.

### SplishSplash

SplishSplash is simple to run and the simulator is simply configured via a few configuration files.
The required one is a `.json` file that contains the parameters of the simulation.
The others are extra input files that determine the geometry and properties of the
objects and fluids in the simulation.

> 

Reviewer: This section requires an introduction to the problem, ie, what are we simulation.
I suggest somethinf like:

```suggestion
### Falling fluid box with [SPlisHSPlasH](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH)

This example uses a straight forward call to the low-level api methods in `Inductiva API`
to simulate a cube of <fluid> falling inside an empty box. The fluid cube has size xxx and
is being dropped into a an empty cubic box with side YYY.  
Input to the simulator require one  `config.json` containing the configuration parameters
of the simulation and one `unit_box.obj` file defining the geometry on a unit cube (used
to describe the shape of both the falling fluid cube and container box).

 This example assumes all input files are inside a single folder, which gets passed to the
 `run` method of a `SPlisHSPlasH` instance as an argument, along with the name of the
 file configuring the simulation.
 Simulations are submitted as tasks to a remote execution queue. A free executor will then
 pick the task from the queue and run it with SPlisHSPlasH. Hence, the user needs to wait for
 the task to finish before retrieving the output files generated by the simulator.
 The following snippets exemplifies how this flow can be achieved.
 

```

> ```python
> import inductiva
> 
> input_dir = inductiva.utils.files.download_from_url(
>     "https://storage.googleapis.com/inductiva-api-demo-files/"
>     "splishsplash-input-example.zip", unzip=True)
> 
> # Set simulation input directory
> splishsplash = inductiva.simulators.SplishSplash()
> 
> task = splishsplash.run(input_dir=input_dir,
>                         sim_config_filename="config.json")
> 
> # Wait for the simulation to finish
> task.wait()
> task.download_outputs()
> ```

Reviewer: ````suggestion

1. Inspect the content of the input folder for this simulation:

```bash
$ ls -l1 sph_demos/simple_splishsplash
config.json
unit_box.obj
```

2. Inspect the content of the python script:

```python
$ cat simple_splishsplash.py
import inductiva

# Instantiate a SPlisHSPlasH proxy
splishsplash = inductiva.simulators.SplishSplash()

# submit the task to the remote execution pipeline, indicating 
# the path to the local directory containing the required input files
# and the name of the configuration file
task = splishsplash.run(input_dir='sph_demos/splishsplash',
                        sim_config_filename="config.json")

# wait for the simulation to finish
task.wait()

# and download all generated output files
task.download_outputs()
```

3. Running the python script:

```bash
$ python simple_splishsplash.py
2023-12-20 14:51:06 INFO Task ID: 1703083865802376474
2023-12-20 14:51:06 INFO Packing input archive to upload to inductiva.ai.
2023-12-20 14:51:06 INFO Uploading packed input archive with size 1.07 KB.
2023-12-20 14:51:06 INFO Input archive uploaded.
2023-12-20 14:51:06 INFO Task submitted to the default resource pool.
2023-12-20 14:51:21 INFO Task completed successfully.
2023-12-20 14:51:30 INFO Downloading simulation outputs to inductiva_output/1703083865802376474/output.zip.
100%|██████████████████████| 261k/261k [00:00<00:00, 5.15MB/s]
2023-12-20 14:51:31 INFO Uncompressing the outputs to inductiva_output/1703083865802376474.
```

4. Inspecting the output of the simulation in `inductiva_output/1703083865802376474`:

```bash
$ ls -l1 inductiva_output/1703083865802376474
log
stderr.txt
stdout.txt
vtk
```

````

As soon you launch the simulation with this script, you will receive the following
logs telling how the simulation is going:

<img src="assets/simulation_logs.png" alt="Simulation Logs">

> ### DualSPHysics

Reviewer: let's do something similar for this case.
Get all input files into a local folder `sph_demos/simple_dualsphysics` and generate a descritption of what the sim is doing.

```python
import inductiva

# Download the configuration files into a folder
input_dir = inductiva.utils.files.download_from_url(
    "https://storage.googleapis.com/inductiva-api-demo-files/"
    "dualsph-duck-input-example.zip", unzip=True)

commands = [
    {"cmd": "gencase floating_duck floating_duck -save:all", "prompts": []},
    {"cmd": "dualsphysics -mdbc floating_duck floating_duck -dirdataout data -svres", "prompts": []},
    {"cmd": "boundaryvtk -loadvtk AutoActual -motiondata floating_duck/data" \ 
            "-savevtk floating_duck/boundary/duck -onlytype:-all,floating", "prompts": []},
    {"cmd": "partvtk -dirin floating_duck/data -savevtk floating_duck/particles/PartAll", "prompts": []},
    {"cmd": "partvtk -dirin floating_duck/data -savevtk floating_duck/particles/PartFluidOut", "prompts": []},
    {"cmd": "isosurface -dirin floating_duck/data -saveiso floating_duck/Surface/surf", "prompts": []}]
    
task = simulator.run(input_dir="FloatingDuck",commands=commands)

# Initialize the Simulator
dualsphysics = inductiva.simulators.DualSPHysics()

# Run simulation with config files in the input directory
task = dualsphysics.run(input_dir=input_dir,

>                         commands=commands)

Reviewer: 2 tasks?

task.wait()
task.download_outputs()
```

> This simulation is what is providing the amazing video of the duck floating in the water!

Reviewer: this sentence is misleading because we don't say how it can be done. Either remove it, or say how it can be done.
